factory-containers/assemble-system-image

```diff
@@ -60,7 +81,86 @@ def get_system_image_from_ci(target: dict, token: str):
     return image_filename.rstrip('.gz')
 
 
-def copy_container_images_to_wic(target: dict, app_image_dir: str, wic_image: str):
+def get_registry_jwt_token(registry_oauth_token, factory, repo, hub_creds):
+    user_pass = '{}:{}'.format(hub_creds['Username'], hub_creds['Secret'])
+
+    headers = {
+        'Authorization': 'Basic ' + base64.b64encode(user_pass.encode()).decode()
+    }
+
+    params = {
+        'service': 'registry',
+        'scope': 'repository:{}/{}:pull'.format(factory, repo)
+    }
+
+    token_req = requests.get(registry_oauth_token, headers=headers, params=params)
+    token_req.raise_for_status()
```

Comment:
might want to raise an HTTPException with the response text like you do up higher in the file. 
~~~
if not target_resp.ok:
        raise requests.exceptions.HTTPError('Failed to get {}: HTTP_{}\n{}'.
                                            format(target_url, target_resp.status_code, target_resp.text))
~~~

Reply:
sometime this gets so tedious i write a single "requests_get" that does that and sets the osftok in the header.

Reply:
in fact- i forgot - https://github.com/foundriesio/ci-scripts/blob/master/helpers.py#L112  - if that helps

Reply:
it helps but not in this specific case since `jobserv_get` sets a specific header (osf-token) while in these three cases headers are different.